cylc/flow/commands.py

```diff
@@ -360,12 +361,24 @@ async def reload_workflow(schd: 'Scheduler'):
         # give commands time to complete
         sleep(1)  # give any remove-init's time to complete
 
-    # reload the workflow definition
-    schd.reload_pending = 'loading the workflow definition'
-    schd.update_data_store()  # update workflow status msg
-    schd._update_workflow_state()
-    LOG.info("Reloading the workflow definition.")
     try:
+        # Back up the current config in case workflow reload errors
+        global_cfg_old = glbl_cfg()
+
+        if reload_global:
+            # Reload global config if requested
+            schd.reload_pending = 'reloading the global configuration'
```

Comment:
The content and use of these `self.reload_pending` strings is a bit messy, but partly historical so I'll leave it be on this branch (and it's of no real consequence).